src/overlays/actors/ovl_Obj_Demo/z_obj_demo.c

```diff
@@ -29,14 +29,90 @@ const ActorInit Obj_Demo_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void ObjDemo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjDemo* this = THIS;
+    s16 temp_v0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Demo/ObjDemo_Init.s")
+    thisx->params &= 0xFF;
+    if ((thisx->params != 0xFF) && (Flags_GetSwitch(globalCtx, thisx->params))) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        if (thisx->shape.rot.x < 0) {
+            this->unk_148 = 1;
+            thisx->shape.rot.x = -thisx->shape.rot.x;
+        }
+        if (thisx->shape.rot.x == 0) {
+            this->unk_14C = 80.0f;
+        } else {
+            this->unk_14C = thisx->shape.rot.x * 4.0f;
+            thisx->shape.rot.x = 0;
+            thisx->world.rot.x = 0;
+        }
+        temp_v0 = thisx->shape.rot.z;
+        if (temp_v0 == 0) {
```

Comment:
https://decomp.me/scratch/Rr4QY can get rid of the temp by moving the assignment up so it uses a different register. Matches the x case further up better too.